Source/WebGPU/WebGPU/WebGPU.h

```diff
@@ -1872,4 +1872,35 @@ WGPU_EXPORT WGPUBool wgpuXRViewIsValid(WGPUXRView view) WGPU_FUNCTION_ATTRIBUTE;
 } // extern "C"
 #endif
 
+#ifdef __cplusplus
+#include <wtf/Platform.h>
+#include <wtf/PlatformEnable.h>
+#include <wtf/Ref.h>
+#include <wtf/RefCounted.h>
+#include <wtf/RetainReleaseSwift.h>
+#if ENABLE(WEBGPU_SWIFT)
+namespace WebGPUTest {
+class CppRefCounted: public RefCounted<CppRefCounted> {
+public:
+    static Ref<CppRefCounted> create() { return adoptRef(*new CppRefCounted()); }
+    // Swift is not importing const methods and RefCount::refCount() is const
+    unsigned refCountProxy() { return refCount(); }
+    WGPU_EXPORT unsigned getSwiftRefCount();
+    WGPU_EXPORT unsigned getSwiftRefCountOptimizedOut();
+} SWIFT_SHARED_REFERENCE(retainCpp, releaseCpp);
+
+}
+
+inline void retainCpp(WebGPUTest::CppRefCounted* obj)
+{
+    WTF::retainRefCounted(obj);
```

Comment:
There are three of these:
1. RefCounted
2. ThreadSafeRefCounted
3. ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr

All three have Global wrappers(which Swift mandates). So I cannot rename all of them as `ref` ?

Also, curious if you are asking me to remove the asserts for perfomance ? I have just added them to make sure I am calling the right `ref`

Reply:
I'm saying something way more basic: If obj is a refcounted WebKit object, it has a member function named 'ref'.

I want to remove the cast because it's not type-safe, and not needed. I want to remove the asserts because they're just there to guard the cast.